packages/editor/src/components/provider/disable-non-page-content-blocks.js

```diff
@@ -45,8 +45,8 @@ export default function DisableNonPageContentBlocks() {
 	);
 	const disabledIds = useSelect( ( select ) => {
 		const { getBlocksByName, getBlockOrder } = select( blockEditorStore );
-		return getBlocksByName( [ 'core/template-part' ] ).flatMap(
-			( clientId ) => getBlockOrder( clientId )
+		return getBlocksByName( 'core/template-part' ).flatMap( ( clientId ) =>
```

Comment:
The problem above remains with `contentOnlyBlockTypes`, which is the result of a filter that may return a new array on every call. Is there any way to fix it in the selector instead, so the user of the selector doesn't need to know about it?

Reply:
Maybe a wrapper `getBlocksByName` shouldn't be memoized, checks for shallow equality, then the underlying function is memoized.

Reply:
The filter is called inside `useMemo,` so I think `contentOnlyBlockTypes` should have a stable ref during the component's life cycle.

> Is there any way to fix it in the selector instead, so the user of the selector doesn't need to know about it?

That's a tricky part with memoizing selectors that accept array or object as arguments. I floated an idea of making them [variadic functions](https://github.com/WordPress/gutenberg/pull/63732#discussion_r1689286859), but I've not tested my theory yet.